crates/ruff_linter/src/settings/mod.rs

```diff
@@ -98,22 +102,8 @@ impl ResolverSettings {
 }
 
 #[derive(Debug, CacheKey)]
-#[allow(clippy::struct_excessive_bools)]
-pub struct Settings {
-    #[cache_key(ignore)]
-    pub cache_dir: PathBuf,
-    #[cache_key(ignore)]
-    pub fix: bool,
-    #[cache_key(ignore)]
-    pub fix_only: bool,
-    #[cache_key(ignore)]
-    pub format: SerializationFormat,
-    #[cache_key(ignore)]
-    pub show_fixes: bool,
-    #[cache_key(ignore)]
-    pub show_source: bool,
-
-    pub resolver: ResolverSettings,
+pub struct LinterSettings {
```

Comment:
The changes in this file are the main changes. Everything else are just downstream refactors to use the new structure.